src/System.Memory/src/System/Buffers/Text/Utf8Formatter/Utf8Formatter.Boolean.cs

```diff
@@ -27,28 +29,77 @@ public static partial class Utf8Formatter
         /// </exceptions>
         public static bool TryFormat(bool value, Span<byte> buffer, out int bytesWritten, StandardFormat format = default)
         {
-            ReadOnlySpan<byte> result;
-            if (format.IsDefault || format.Symbol == 'G')
-            {
-                result = value ? Utf8Constants.s_capitalizedTrue : Utf8Constants.s_capitalizedFalse;
-            }
-            else if (format.Symbol == 'l')
+            char symbol = FormattingHelpers.GetSymbolOrDefault(format, 'G');
+
+            if (value)
             {
-                result = value ? Utf8Constants.s_true : Utf8Constants.s_false;
+                if (symbol == 'G')
+                {
+                    // By having each branch perform its own call to TryWriteUInt32BigEndian, we ensure that a
+                    // constant value is passed to this routine, which means the compiler can reverse endianness
+                    // at compile time instead of runtime if necessary.
+                    const uint TrueValueUppercase = ('T' << 24) + ('r' << 16) + ('u' << 8) + ('e' << 0);
+                    if (!BinaryPrimitives.TryWriteUInt32BigEndian(buffer, TrueValueUppercase))
+                    {
+                        goto BufferTooSmall;
+                    }
+                }
+                else if (symbol == 'l')
+                {
+                    const uint TrueValueLowercase = ('t' << 24) + ('r' << 16) + ('u' << 8) + ('e' << 0);
+                    if (!BinaryPrimitives.TryWriteUInt32BigEndian(buffer, TrueValueLowercase))
```

Comment:
Would writing TryFormat using BinaryPrimitives still be necessary once TryCopyTo is optimized?
https://github.com/dotnet/coreclr/issues/15076

Reply:
cc @GrabYourPitchforks